### ISSUE TITLE
"Operation returned an invalid status: Not Acceptable" issue raised while using generatedetailedCost report from azure.mgmt.costmanagement SDK

### ISSUE THREAD

**Issue by tsaindane-cloudwiz** [Azure/azure-sdk-for-python#34096]
- **Package Name**: 
- **Package Version**: 
- **Operating System**:
- **Python Version**: 

**Describe the bug**
We are using azure.mgmt.costmanagement python SDK and when we tried to begin_create_operation() from generateDetailedCostReport class then ' Operation returned an invalid status 'Not Acceptable'' this error came up. And this error is not frequently comes up.

**To Reproduce**
1. 

**Expected behavior**
This error should not happen.

**Screenshots**
![Screenshot 2024-02-01 at 2 43 19 PM](https://github.com/Azure/azure-sdk-for-python/assets/81921980/496e4341-425c-42a8-864d-1b38f2be2949)

**Additional context**

**Comment by l0lawrence** [comment]
Hi @tsaindane-cloudwiz  thank you for the feedback, can you please provide the package-version, operating system and python-version and we will get back to you asap.

**Comment by tsaindane-cloudwiz** [comment]
python package version: 4.0.1
environment: google cloud function V1
python version: 3.9

**Comment by HuyNguyen7994** [comment]
I'm having the same problem. Running the code below returns `HttpResponseError: Operation returned an invalid status 'Not Acceptable'`

```python
from azure.identity import EnvironmentCredential
from azure.mgmt.costmanagement import CostManagementClient, models
from dotenv import load_dotenv

load_dotenv()

client = CostManagementClient(credential=EnvironmentCredential())

time_period = models.CostDetailsTimePeriod(start="2024-01-01", end="2024-12-31")
gen_cost_details_def = models.GenerateCostDetailsReportRequestDefinition(metric="ActualCost", time_period=time_period)

response = client.generate_cost_details_report.begin_create_operation(scope="/subscriptions/xxxxx-xxxx-xxxx-xxxx-xxxx", parameters=gen_cost_details_def).result()

print(response)
```

Other information:

```
python: 3.12.7
environment: macOS Sequoia 15.3
azure.mgmt.costmanagement: 4.0.1
```

@tsaindane-cloudwiz was you able to solve this?

**Comment by tsaindane-cloudwiz** [comment]
@HuyNguyen7994 No, I did not found any solution/ workaround for this. But now its not happening for me after then. I am still monitoring.

**Comment by msyyc** [comment]
Please reopen this issue if needed.
